### DISCUSSION TITLE
Turbo Module without Codegen

### DISCUSSION THREAD

**Discussion by anshulnoori** [reactwg/react-native-new-architecture#162] [Q&A]
In the [docs for the New Architecture](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/codegen.md), it mentions:
> Using Codegen is not mandatory -- all the code that is generated by it can also be written manually.

However, I cannot find any resources relating to creating a Turbo Module without using Codegen. Where can I find the resources relating to this (if they exist)?

**Comment by cortinico** [comment]
> I cannot find any resources relating to creating a Turbo Module without using Codegen. Where can I find the resources relating to this (if they exist)?

The easiest way to achieve this is to setup your project to run codegen once, and copy the generated code in your project. We currently don't have any documentation related to this flow as the reccomendation is to use codegen (as if you copy over the generated code, it won't follow through with future improvements we'll be shipping).
